network-libp2p/tests/discovery.rs

```diff
@@ -77,10 +72,20 @@ impl TestNode {
         let peer_contact_book = Arc::new(RwLock::new(PeerContactBook::new(peer_contact)));
 
         let clock = Arc::new(OffsetTime::new());
-        let behaviour =
-            DiscoveryBehaviour::new(config, keypair, Arc::clone(&peer_contact_book), clock);
+        let behaviour = discovery::Behaviour::new(
+            config,
+            keypair.clone(),
+            Arc::clone(&peer_contact_book),
+            clock,
+        );
 
-        let mut swarm = Swarm::with_threadpool_executor(transport, behaviour, peer_id);
+        let mut swarm = SwarmBuilder::with_existing_identity(keypair)
```

Comment:
There is also `libp2p-swarm-test` which allows for quick, in-memory testing of `Swarm`s.